aten/src/ATen/native/cuda/AdaptiveMaxPooling2d.cu

```diff
@@ -77,7 +77,7 @@ __global__ void adaptivemaxpool(T *input, T *output, int64_t *indices,
       T *ptr_output = output + oh*osizeW + ow;
       int64_t *ptr_ind = indices + oh*osizeW + ow;
       int argmax = -1;
-      T max = THCNumerics<T>::min();
+      T max = at::numeric_limits<T>::lower_bound(); // -Infinity
```

Comment:
If there are only -inf in the pooling window, argmax will not be assigned, and there will be invalid memory access in backward. Your tests don't test backward.

Reply:
Fixed. `argmax` is now initialized with a value similar to those in the loop.